pallets/rc-migrator/src/staking/bags_list.rs

```diff
@@ -0,0 +1,179 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Nomination pools data migrator module.
+
+use crate::{types::*, *};
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub enum BagsListStage<AccountId, Score> {
+	ListNodes(Option<AccountId>),
+	ListBags(Option<Score>),
+	Finished,
+}
+
+pub type BagsListStageOf<T> = BagsListStage<
+	<T as frame_system::Config>::AccountId,
+	<T as pallet_bags_list::Config<pallet_bags_list::Instance1>>::Score,
+>;
+
+#[derive(
+	Encode,
+	Decode,
+	MaxEncodedLen,
+	TypeInfo,
+	RuntimeDebugNoBound,
+	CloneNoBound,
+	PartialEqNoBound,
+	EqNoBound,
+)]
+#[codec(mel_bound(T: Config))]
+#[scale_info(skip_type_params(T))]
+pub enum RcBagsListMessage<T: pallet_bags_list::Config<pallet_bags_list::Instance1>> {
+	Node { id: T::AccountId, node: alias::NodeOf<T> },
+	Bag { score: T::Score, bag: alias::BagOf<T> },
+}
+
+pub struct BagsListMigrator<T> {
+	_phantom: PhantomData<T>,
+}
+
+impl<T: Config> PalletMigration for BagsListMigrator<T> {
+	type Key = BagsListStageOf<T>;
+	type Error = Error<T>;
+
+	fn migrate_many(
+		current_key: Option<Self::Key>,
+		weight_counter: &mut WeightMeter,
+	) -> Result<Option<Self::Key>, Self::Error> {
+		let mut inner_key = current_key.unwrap_or(BagsListStage::ListNodes(None));
+		let mut messages = Vec::new();
+
+		loop {
+			if weight_counter
+				.try_consume(<T as frame_system::Config>::DbWeight::get().reads_writes(1, 1))
+				.is_err()
+			{
+				if messages.is_empty() {
+					return Err(Error::OutOfWeight);
+				} else {
+					break;
+				}
+			}
+			if messages.len() > 10_000 {
```

Comment:
temporary measure or not? can you explain please

Reply:
I was not sure here how exactly we benchmark things. Depending that, this would be a failsafe in case that our weights are wrong to not completely overload the system.  
It is not great or precise, but better than some infinite loop.